activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -356,7 +356,11 @@ def find_with_associations
 
     def construct_join_dependency(joins = [])
       including = eager_load_values + includes_values
-      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins)
+      finder_methods_hash = {
+        eager_load: eager_load_values,
+        includes: includes_values
+      }
+      ActiveRecord::Associations::JoinDependency.new(@klass, including, joins, finder_methods_hash)
```

Comment:
Hm, I agree that creating and passing the `including` variable is now unnecessary. But since `ActiveRecord::Associations::JoinDependency` is public and I can see other repos on Github using and initializing it directly, changing the `initialize` method signature would break backwards compatibility. Not sure what to do in this case?